lib/spack/spack/installer.py

```diff
@@ -1702,6 +1704,29 @@ def _install_task(self, task: BuildTask, install_status: InstallStatus) -> None:
             # If a compiler, ensure it is added to the configuration
             if task.compiler:
                 self._add_compiler_package_to_config(pkg)
+
+            # Iterate over all mirrors with autopush==True
```

Comment:
I guess you're putting this here instead of in a post-install hook because it relies on the package being installed in the database, correct?

I'd rather see this in a new post-install hook, cause the installer is already doing (too) many things.

(I've created an issue for `pkg.windows_establish_runtime_linkage()` which certainly does not belong there: https://github.com/spack/spack/issues/42445, similarly the `if task.compiler` bits do not belong there.

Sounds like we need yet another hook that runs after a package is added to the database.

I think we can actually repurpose

```spack.hooks.on_install_success(self.pkg.spec)```

for this, cause as far as I can see, nobody implements that hook.

Reply:
I am putting it in the installer because of my limited knowledge of Spack (I did not know about the hooks). And yes, as it relies on the package being installed (compiled) it seemed logical to put it there.

Thanks for your suggestions, I am going to look at the hooks.

Reply:
Perfect! You can also reach out on Slack if you have questions of why or where things are done.